### PR TITLE
Rewrite file-vault urls to remove AWS args

### PR DESCRIPTION
**Pull request by easternbloc**
This stops the URL's being returned by file-vault containing the same params as the signedUrl that's returned from AWS.

### CODE REVIEW

File: controllers/file.js [hunk 1]
```diff
@@ -9,6 +9,11 @@ const fs = require('fs');
 const onFinished = require('on-finished');
 const config = require('config');
 const path = require('path');
+const {URL} = require('url');
+
+const crypto = require('crypto');
+const algorithm = 'aes-256-ctr';
+const password = config.get('aws.password');
```
Comment (lennym) [inline controllers/file.js]: You _might_ want to consider throwing an error if the password is `changeme`. Bullet proof socks 'n' all that.

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM node:10-alpine
+FROM node:8-alpine
```
Comment (lennym) [inline Dockerfile]: How come this?

**Comment by lennym** [comment]
The idea here being that instead of exposing the AWS parameters in the returned URL, it encrypts them with a password to make a "random" parameter, which when it receives back in a request it decrypts and proxies through to S3 with the decrypted AWS parameters?

So the AWS signature is never exposed in cleartext outside of the runtime of this service?

File: controllers/file.js (hunk 1 above)
Comment (easternbloc) [reply controllers/file.js]: will do exactly that 👍

**Comment by easternbloc** [comment]
yes exactly that! I'll make the changes to the password issues.

File: Dockerfile (hunk 2 above)
Comment (easternbloc) [reply Dockerfile]: it was always broken in 10 i only just spotted it here. I'll fix the 10 issues in an upcoming PR

**Pull request merged**